compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -46,14 +61,115 @@ class ExtractSemanticDB extends Phase:
   // Check not needed since it does not transform trees
   override def isCheckable: Boolean = false
 
-  override def run(using Context): Unit =
-    val unit = ctx.compilationUnit
-    val extractor = Extractor()
-    extractor.extract(unit.tpdTree)
-    ExtractSemanticDB.write(unit.source, extractor.occurrences.toList, extractor.symbolInfos.toList, extractor.synthetics.toList)
+  override def runOn(units: List[CompilationUnit])(using ctx: Context): List[CompilationUnit] = {
+    val appendDiagnostics = phaseMode == ExtractSemanticDB.PhaseMode.AppendDiagnostics
+    val warnings =
+      if (appendDiagnostics)
+        ctx.reporter.allWarnings.groupBy(w => w.pos.source)
+      else Map.empty
+
+    units.asJava.parallelStream().forEach { unit =>
```

Comment:
see my comment when this was `.map` but yeah its not safe to parallelise things that use the `Context`

Reply:
Oops, thanks! fixed in https://github.com/lampepfl/dotty/pull/17835/commits/50e0fd2eacafb58453a60e52a4f43a53b0c8e02d